pallets/nft/src/lib.rs

```diff
@@ -281,71 +223,145 @@ pub mod pallet {
 	}
 
 	#[pallet::error]
 	pub enum Error<T> {
-		/// The class does not exist
-		ClassNotFound,
-		/// The token does not exist
-		TokenNotFound,
-		/// Not the class owner
-		NotClassOwner,
-		/// Not the token owner
-		NotTokenOwner,
-		/// The token class is not empty
-		NonZeroIssuance,
-		/// Token is currently locked
-		TokenLocked,
-		/// A token can not be transferred to self
-		CannotSendToSelf,
-		/// A user cannot buy already owned token
-		CannotBuyOwnToken,
-		/// Token has been already bought from a pool
-		TokenAlreadyHasAnOwner,
-		/// A token still owned by class owner
-		CannotSellPoolToken,
-		/// Class wasn't created as a pool
-		NotAPool,
-		/// Metadata exceed the allowed length
-		MetadataTooLong,
+		/// Count of instances overflown
+		NoAvailableInstanceId,
+		/// Count of classes overflown
+		NoAvailableClassId,
+		/// Class still contains minted tokens
+		TokenClassNotEmpty,
+		/// Class does not exist
+		ClassUnknown,
+		/// Shares has to be set for liquidity mining
+		SharesNotSet,
+		/// Operation not permitted
+		NotPermitted,
 	}
 }
 
 impl<T: Config> Pallet<T> {
-	pub fn is_owner(account: &T::AccountId, token: (T::ClassId, T::TokenId)) -> bool {
-		orml_nft::Pallet::<T>::is_owner(account, token)
+	pub fn get_next_class_id() -> Result<T::NftClassId, Error<T>> {
+		NextClassId::<T>::try_mutate(|id| {
+			let current_id = *id;
+			*id = id.checked_add(&One::one()).ok_or(Error::<T>::NoAvailableClassId)?;
+			Ok(current_id)
+		})
+	}
+
+	pub fn get_next_instance_id(class_id: T::NftClassId) -> Result<T::NftInstanceId, Error<T>> {
+		NextInstanceId::<T>::try_mutate(class_id, |id| {
+			let current_id = *id;
+			*id = id.checked_add(&One::one()).ok_or(Error::<T>::NoAvailableInstanceId)?;
+			Ok(current_id)
+		})
 	}
 
-	pub fn is_locked(token: (T::ClassId, T::TokenId)) -> Result<bool, DispatchError> {
-		let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
-		Ok(token_info.data.locked)
+	pub fn class_owner(class_id: T::NftClassId) -> Option<T::AccountId> {
+		pallet_uniques::Pallet::<T>::class_owner(&class_id.into())
 	}
 
-	pub fn toggle_lock(account: &T::AccountId, token_id: (T::ClassId, T::TokenId)) -> DispatchResult {
-		let _class_info = orml_nft::Pallet::<T>::classes(token_id.0).ok_or(Error::<T>::ClassNotFound)?;
-		orml_nft::Tokens::<T>::mutate_exists(token_id.0, token_id.1, |token| -> DispatchResult {
-			if let Some(ref mut token) = token {
-				ensure!(*account == token.owner, Error::<T>::NotTokenOwner);
-				token.data.locked ^= true; // Toggle
-				Self::deposit_event(Event::TokenMintedLockToggled(
-					account.clone(),
-					token_id.0,
-					token_id.1,
-					token.data.locked,
-				));
-			}
-			Ok(())
-		})
+	pub fn instance_owner(class_id: T::NftClassId, instance_id: T::NftInstanceId) -> Option<T::AccountId> {
+		pallet_uniques::Pallet::<T>::owner(class_id.into(), instance_id.into())
+	}
+
+	pub fn do_create_class(
```

Comment:
i would prefer to allow specifying your own id when creating class, we can have some range reserved for runtime nfts